pkg/apis/servicecatalog/v1alpha1/types.go

```diff
@@ -409,8 +409,43 @@ type ServiceInstance struct {
 	Status ServiceInstanceStatus `json:"status"`
 }
 
-// ServiceInstanceSpec represents the desired state of an Instance.
-type ServiceInstanceSpec struct {
+// DesiredPlan defines the user specification for the desired
+// ServicePlan and ServiceClass. Because there are multiple ways to
+// specify the desired Class/Plan, this structure specifies the
+// allowed ways to specify the intent.
+type DesiredPlan struct {
```

Comment:
I have a strong desire for this approach.  It will be easiest to validate in the long term if we group the fields on the same object.  We can perform further factoring in the future if it becomes a problem.